src/main/java/org/folio/spring/controller/TenantController.java

```diff
@@ -34,40 +34,24 @@
   public ResponseEntity<Void> postTenant(@Valid TenantAttributes tenantAttributes) {
     if (isDisableJob(tenantAttributes)) {
       log.info("Disabling tenant...");
-      disableTenant();
+      tenantService.deleteTenant(tenantAttributes);
     } else {
       log.info("Upgrading tenant...");
-      upgradeTenant();
+      tenantService.createOrUpdateTenant(tenantAttributes);
       var parameters = tenantAttributes.getParameters();
       var loadReferenceParam = getParameterValue(LOAD_REFERENCE_PARAM, parameters);
       if (loadReferenceParam.isPresent() && loadReferenceParam.get().equals("true")) {
-        loadReferenceData();
+        tenantService.loadReferenceData();
```

Comment:
Agree, these should be moved to the service

Reply:
:+1: